### PR TITLE
gh-100220: Fix error handling in make rules

### PR DESCRIPTION
**Pull request by mgorny**
Use `set -e` before compound shell commands in order to ensure that make targets fail correctly when at least one of the subcommands fail.

This is necessary since make considers a target failed only if one of the shell invocations returns with unsuccessful exit status.  If a shell script does not exit explicitly, the shell uses the exit status of the *last* executed command.  This means that when multiple commands are executed (e.g. through a `for` loop), the exit statuses of prior command invocations are ignored.

This can be either resolved by adding an explicit `|| exit 1` to every command that is expected to succeed, or by running the whole script with `set -e`.  The latter was used here as it the rules seem to be written with the assumption that individual commands were supposed to cause the make rules to fail.

* Issue: gh-100220

### CODE REVIEW

**Comment by zware** [comment]
This looks good, but is going to need some conflict resolution now after GH-100329.

I wonder if this could be achieved with less churn by setting `SHELL := $(SHELL) -e`?

**Comment by mgorny** [comment]
> This looks good, but is going to need some conflict resolution now after [GH-100329](https://github.com/python/cpython/pull/100329).

Done.

> I wonder if this could be achieved with less churn by setting `SHELL := $(SHELL) -e`?

Hmm, I haven't though of that. I suppose it could work too.

**Comment by merwok** [comment]
Please use regular merges and pushes for Python repos, force-pushes give reviewers a poor experience on github.

**Comment by mgorny** [comment]
@zware, do you want to rework this to try updating this to use `SHELL` right now or is this good as-is?

**Comment by zware** [comment]
If the `SHELL` method works and doesn't break anything else, I'd rather go that route.  That way we don't have to worry about making sure new commands start with the right magic incantation, and the history is much cleaner.

**Comment by mgorny** [comment]
Ping.

File: Misc/NEWS.d/next/Build/2022-12-18-07-24-44.gh-issue-100220.BgSV7C.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix error handling in compound Make rules that lead to the install targets
+succeeding when the respective files were not installed.
```
Comment (zware) [inline Misc/NEWS.d/next/Build/2022-12-18-07-24-44.gh-issue-100220.BgSV7C.rst]:
```suggestion
Changed the default value of the ``SHELL`` Makefile variable from ``/bin/sh``
to ``/bin/sh -e`` to ensure that complex recipes correctly fail after an error.
Previously, ``make install`` could fail to install some files and yet return
a successful result.
```

**Review by zware — APPROVED** [review]

**Comment by mgorny** [comment]
Thank you!

**Pull request merged**